core/storage/user/gae_models.py

```diff
@@ -384,6 +384,28 @@ def get_or_create(cls, user_id):
             entity = cls(id=user_id)
         return entity
 
+    @staticmethod
+    def export_data(user_id):
+        """(Takeout) Export the user-relevant properties of UserStatsModel.
+
+        Args:
+            user_id: str. The user_id denotes which user's data to extract.
+
+        Returns:
+            dict. The user-relevant properties of UserStatsModel in a python
+                dict format.
+        """
+        user_model = UserStatsModel.get_or_create(user_id)
+        user_data = {
+            'impact_score': user_model.impact_score,
+            'total_plays': user_model.total_plays,
+            'average_ratings': user_model.average_ratings,
+            'num_ratings': user_model.num_ratings,
+            'weekly_creator_stats_list': user_model.weekly_creator_stats_list
```

Comment:
Per https://docs.google.com/document/d/1WSJYujTTGUADFakMD_-cpAuWZI4yWakL7cXOT3DMx7w/edit#bookmark=id.xfw7p3frtru8 I see that we're following the same format roughly, but I suggest we actually explicitly assemble the weekly_creator_stats_list to guard against accidentally pulling in changes to the underlying structure that we might not want to export to users.

WDYT?

Reply:
I'm okay with explicitly assembling the weekly_creator_stats_list. There might be a varying amount of items in that list depending on when the user starting creator/contributing to explorations, so I wanted to confirm what I would be doing differently in manually constructing the list. In constructing the list, should I be verifying that each object is of the expected form? So verifying types with **type(obj)** and double checking key-names (as it adheres to https://docs.google.com/document/d/1WSJYujTTGUADFakMD_-cpAuWZI4yWakL7cXOT3DMx7w/edit?ts=5c89a9c3#). What should I do if the list contains invalid data? Raise an error or simply not include that item when manually creating the list? Thanks. @BenHenning PTAL

Reply:
I think raising an error is fine--if Takeout fails due to such an error, we should investigate it. Hopefully our future higher-level Takeout tests help prevent these types of situations, but failing here makes sense.

I didn't think about doing the type validation, and I'm not convinced we necessarily need it. I was more concerned with what labeled data we're including, not the contents of that data. We can't verify everything, but if we trust the labels to be correct, then manually reassembling the whole structure makes it really clear what we're choosing to include in Takeout data. Specifically, I see Takeout as an opt-in mechanism for data exporting, not opt-out. Maybe @seanlip has a different view here.

I also added one more comment hoping to clarify one of the bits of information we're exporting here.

Reply:
What @BenHenning says sounds fine to me in all respects (thanks Ben)!